llvm/lib/Target/RISCV/GISel/RISCVInstructionSelector.cpp

```diff
@@ -405,6 +410,49 @@ bool RISCVInstructionSelector::select(MachineInstr &MI) {
     MI.eraseFromParent();
     return constrainSelectedInstRegOperands(*Bcc, TII, TRI, RBI);
   }
+  case TargetOpcode::G_BRJT: {
+    unsigned EntrySize =
```

Comment:
In the long term, it probably makes sense to move this to a lowering action on the legalizer.

Reply:
X86 uses Has variables:
https://github.com/llvm/llvm-project/blob/0448a1c0dc131835301a59db7138d504b836a591/llvm/lib/Target/X86/GISel/X86InstructionSelector.cpp#L444

Reply:
@tschuett I'm not sure what you're suggesting I do in this patch?

Reply:
If I read your comments correctly, you have different options for the code sequence depending on which extensions are available. X86 has the same loads of 256-bit are legal, but depending on pure AVX or with VLX they haven selection strategies.

Reply:
The load size is determined by codemodel and pic/no-pic. This is reflected in EntrySize and the isPositionIndependent. This also affects the jump-table kind you see in the tests `custom32`, `block-address`, or `label-difference32`

The SLLI+ADD can be fused to SH2ADD or SH3ADD with the Zbb extension.

I think we can expand this whole thing to generic opcodes G_SEXTLOAD, G_SHL, G_ADD, G_BRINDIRECT, etc. during legalization and let all the pieces go through regbankselect and instruction-select individually. But G_BRINDIRECT hasn't been implemented yet. I have more fallbacks pointing at missing G_BRJT/G_JUMP_TABLE support so I wrote this to make progress.